File: testprojects/scala_main/pipelines/SCALA_DEP_MGMT_ALL/code/src/main/scala/org/main/scla_dep_mgmt/graph/Reformat_5.scala
package org.main.scla_dep_mgmt.graph

import io.prophecy.libs._
import org.main.scla_dep_mgmt.config.ConfigStore._
import org.main.scla_dep_mgmt.config.Context
import org.main.scla_dep_mgmt.udfs.UDFs._
import org.main.scla_dep_mgmt.udfs._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object Reformat_5 {

  def apply(context: Context, in: DataFrame): DataFrame =
    in.select(
      col("col1"),
      col("col2"),
      col("c3"),
      col("c8_c1"),
      col("c8_c2"),
      col("c9_udf1_c1"),
      col("c9_udf1_c2"),
      substring(col("c9_udf1_c2"), 1, 2).cast(IntegerType).as("cast_c9_sql"),
      lit(context.config.c_sql_c8c1).as("c_sql_c8c1_sql")
    )

}

File: testprojects/python_main/pipelines/PYTHON_SG_SRC/code/com/sg_src/main/graph/all_type_main_pythonsg/Subgraph_4_1_1/Subgraph_1/Reformat_5.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.libs import typed_lit
from .config import *
from com.sg_src.main.udfs.UDFs import *

def Reformat_5(spark: SparkSession, in0: DataFrame) -> DataFrame:
    return in0.select(
        concat(lit(Config.c_sg1_c_string), col("`c-string`")).alias("c1"), 
        col("c_udfs_usage"), 
        col("`c-string`")
    )
